react/components/IngredientsContent.js

```diff
@@ -45,56 +48,58 @@ class IngredientsContent extends Component {
           <FormattedMessage id="product-customizer.change-ingredients" />
         </h3>
         <div className="change-ingredients--selected-variation">
-          <legend className="bg-near-white w-100 ph2 pv4 mb4">
-            <FormattedMessage id="product-customizer.select-your-ingredients" />
-          </legend>
-          <div className="change-ingredients--selected-ingredients ph2 mb5">
-            <p className="pv3 bb b--light-gray ttu ma0 f7 b near-black">
-              <FormattedMessage id="product-customizer.selected-ingredients" />
-            </p>
-            <ul className="ma0 pa0 list">
-              {compositionVariations.variations.map((ingredient, key) => (
-                <li key={key}>
-                  <ToggledChoice
-                    item={ingredient}
-                    minTotalItems={compositionVariations.minTotalItems}
-                    maxTotalItems={compositionVariations.maxTotalItems}
-                    index={ingredient.name}
-                    chosenAmount={chosenAmountBasic}
-                    onVariationChange={onVariationChangeBasic}
-                  />
-                </li>
-              ))}
-            </ul>
-          </div>
-          <div className="change-ingredients--extra-ingredients ph2 mb5">
-            <p className="pv3 bb b--light-gray ttu ma0 f7 b near-black">
-              <FormattedMessage id="product-customizer.extra-ingredients" />
-            </p>
-            <ul className="ma0 pa0">
-              {optionalVariations.variations.map((ingredient, key) => (
-                <li
-                  key={key}
-                  className={classnames(
-                    ['flex', 'justify-between', 'items-center', 'pv4'],
-                    {
-                      ['bb b--light-gray']:
-                        key !== optionalVariations.variations.length - 1,
-                    }
-                  )}
-                >
-                  <MultipleChoice
-                    item={ingredient}
-                    minTotalItems={optionalVariations.minTotalItems}
-                    maxTotalItems={optionalVariations.maxTotalItems}
-                    index={ingredient.name}
-                    chosenAmount={chosenAmount}
-                    onVariationChange={onVariationChange}
-                  />
-                </li>
-              ))}
-            </ul>
-          </div>
+          {this.hasVariations(compositionVariations) && (
```

Comment:
Is there a way that `variations == []`?

Reply:
No, because the schema will not generate if the attachment does not conform to a regex, and that regex forces having at least one SKU ID and its configuration.
But I can add a verification for empty array if necessary, just in case.

Reply:
#done
https://github.com/vtex-apps/product-customizer/pull/10/commits/18370361f06a2a92e749690c4fbcc9a3521df33b